lblod/frontend-gelinkt-notuleren
# Update to ember-modifier 4.1 and fix some deprecations

### Overview
Doesn't fix all the deprecations from ember-data or even the one from AuLoader but `async: true` was the worst culprit and could be done completely brainlessly. The need to change ember-modifier peer-deps was the only part that took time.
Draft until a new editor release happens.

##### connected issues and PRs:
N/A

### Setup
N/A

### How to test/reproduce
This should disable at least most of the other ember-data deprecations to give you the zen developer experience you've always wanted (and to check that I've got rid of the async and modifier ones) in app/app.js:
```
import { registerDeprecationHandler } from '@ember/debug';

registerDeprecationHandler((message, options, next) => {
  if (
    [
      'The `firstObject` method on the class ManyArray is deprecated. Use the native array method `[0]` instead.',
      'The `firstObject` method on the class RecordArray is deprecated. Use the native array method `[0]` instead.',
      'The `toArray` method on the class RecordArray is deprecated. Use the native array method `slice` instead.',
      'The `toArray` method on the class ManyArray is deprecated. Use the native array method `slice` instead.',
      'The `sortBy` method on the class RecordArray is deprecated. Use the native array method `.slice().sort` instead.',
      'The `sortBy` method on the class ManyArray is deprecated. Use the native array method `.slice().sort` instead.',
      'Do not use A() on an EmberData PromiseManyArray',
      'The `pushObject` method on the class ManyArray is deprecated. Use the native array method `push` instead.',
      'The `pushObjects` method on the class ManyArray is deprecated. Use the native array method `push` instead.',
      "The pushObject method on ember-data's PromiseManyArray is deprecated. await the promise and work with the ManyArray directly.",
      "The map method on ember-data's PromiseManyArray is deprecated. await the promise and work with the ManyArray directly.",
      'The `setObjects` method on the class ManyArray is deprecated. Use the native array method ``arr.length = 0; arr.push(objects);`` instead.',
      'The `clear` method on the class ManyArray is deprecated. Use the native array method `length = 0` instead.',
      "The toArray method on ember-data's PromiseManyArray is deprecated. await the promise and work with the ManyArray directly.",
    ].includes(message) ||
    message.startsWith(
      '[AuLoader] This version of the `AuLoader` component is deprecated.',
    )
  ) {
    return;
  }
  console.log(message);
  return next(message, options);
});
```

### Challenges/uncertainties
The further deprecation fixes won't be hard, just require some second thought as to whether they're doing the right thing.

### Checks PR readiness
